js_modules/dagit/packages/core/src/ui/SectionedLeftNav.tsx

```diff
@@ -0,0 +1,267 @@
+import {BaseTag, Box, Colors, Icon, IconWrapper, StyledTag} from '@dagster-io/ui';
+import * as React from 'react';
+import styled from 'styled-components/macro';
+
+import {AppContext} from '../app/AppContext';
+import {isAssetGroup} from '../asset-graph/Utils';
+import {useStateWithStorage} from '../hooks/useStateWithStorage';
+import {getJobItemsForOption, JobItem} from '../nav/FlatContentList';
+import {DagsterRepoOption, WorkspaceContext} from '../workspace/WorkspaceContext';
+import {buildRepoAddress} from '../workspace/buildRepoAddress';
+import {repoAddressAsString} from '../workspace/repoAddressAsString';
+import {RepoAddress} from '../workspace/types';
+
+const validateExpandedKeys = (parsed: unknown) => (Array.isArray(parsed) ? parsed : []);
+const EXPANDED_REPO_KEYS = 'dagit.expanded-repo-keys';
+
+export const SectionedLeftNav = () => {
+  const {loading, visibleRepos} = React.useContext(WorkspaceContext);
+  const {basePath} = React.useContext(AppContext);
+
+  const [expandedKeys, setExpandedKeys] = useStateWithStorage<string[]>(
+    basePath + ':' + EXPANDED_REPO_KEYS,
```

Comment:
It'd be interesting to explore baking this prefixing into useStateWithStorage. I can't think of any cases where this would be detrimental and I don't think I've been doing it...